conn_go18.go

```diff
@@ -44,13 +45,30 @@ func (cn *conn) ExecContext(ctx context.Context, query string, args []driver.Nam
 
 // Implement the "ConnBeginTx" interface
 func (cn *conn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx, error) {
-	if opts.Isolation != 0 {
-		return nil, errors.New("isolation levels not supported")
+	mode := ""
+
+	switch sql.IsolationLevel(opts.Isolation) {
+	case sql.LevelDefault:
+		// Don't touch mode: use the server's default
+	case sql.LevelReadUncommitted:
+		mode += " ISOLATION LEVEL READ UNCOMMITTED"
+	case sql.LevelReadCommitted:
+		mode += " ISOLATION LEVEL READ COMMITTED"
+	case sql.LevelRepeatableRead:
+		mode += " ISOLATION LEVEL REPEATABLE READ"
+	case sql.LevelSerializable:
+		mode += " ISOLATION LEVEL SERIALIZABLE"
+	default:
+		return nil, fmt.Errorf("isolation level not supported: %d", opts.Isolation)
 	}
+
 	if opts.ReadOnly {
-		return nil, errors.New("read-only transactions not supported")
+		mode += " READ ONLY"
+	} else {
+		mode += " READ WRITE"
```

Comment:
This can be changed via server config (`default_transaction_read_only`) and set per-session, so it's probably best if the library makes no assumptions about defaults.